### PR TITLE
Add Design Philosophy

### PR DESCRIPTION
**Pull request by svekars**

### CODE REVIEW

**Review by albanD — APPROVED** [review]
SGTM

File: docs/source/community/design.rst [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+PyTorch core library, but ultimately benefit from, as evidenced by
+our `rich ecosystem <https://pytorch.org/ecosystem/>`__. In other
+words, not automating at the start allows us to potentially reach levels
+of good automation faster.
+
+Principle 3: Python First with Best In Class Language Interoperability
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This principle began as `Python
+First <http://This%20link%20seems%20broken>`__:
```
Comment (svekars) [inline docs/source/community/design.rst:118]: This link was broken in the Google doc. Can we remove it and just keep "Python First" w/o link?

Comment (orionr) [reply docs/source/community/design.rst:118]: Agreed - let's do that. Thanks!

**Review by orionr — COMMENTED** [review]
Can we also fix the lint errors with trailing spaces?

**Pull request closed without merging**